controllers/apps/v1beta3/emqx_handler.go

```diff
@@ -943,3 +933,65 @@ func mergeEnvAndConfig(instance appsv1beta3.Emqx) (ret []corev1.EnvVar) {
 
 	return
 }
+
+func isPluginInitialized(instance appsv1beta3.Emqx) bool {
+	// Init Plugin
+	for _, c := range instance.GetStatus().Conditions {
+		if c.Type == appsv1beta3.ConditionPluginInitialized && c.Status == corev1.ConditionTrue {
+			return true
+		}
+	}
+	return false
+}
+
+func generateReloaderContainer(r *EmqxReconciler, instance appsv1beta3.Emqx, sts *appsv1.StatefulSet) corev1.Container {
+	username, password, apiPort := r.getManagementField(instance)
+	exist, container := findEmqxContainer(sts.Spec.Template.Spec.Containers)
+	if !exist {
+		return corev1.Container{}
+	}
+	return corev1.Container{
+		Name:            ReloaderContainerName,
+		Image:           ReloaderContainerImage,
+		ImagePullPolicy: instance.GetImagePullPolicy(),
+		Env:             container.Env,
+		VolumeMounts:    container.VolumeMounts,
+		Args: []string{
+			"-u", username,
+			"-p", password,
+			"-P", apiPort,
+		},
+	}
+}
+
+func generateEmqxContainer(instance appsv1beta3.Emqx) corev1.Container {
+	return corev1.Container{
+		Name:            EmqxContainerName,
+		Image:           instance.GetImage(),
+		ImagePullPolicy: instance.GetImagePullPolicy(),
+		Resources:       instance.GetResource(),
+		Env:             mergeEnvAndConfig(instance),
+		Args:            instance.GetArgs(),
+		ReadinessProbe:  instance.GetReadinessProbe(),
+		LivenessProbe:   instance.GetLivenessProbe(),
+		StartupProbe:    instance.GetStartupProbe(),
+	}
+}
+
+func findEmqxContainer(containers []corev1.Container) (bool, corev1.Container) {
+	for _, c := range containers {
+		if c.Name == EmqxContainerName {
+			return true, c
+		}
+	}
+	return false, corev1.Container{}
+}
+
+func setadditionalAnnotations(sts *appsv1.StatefulSet) {
+	if sts.Annotations == nil {
+		sts.Annotations = map[string]string{}
+	}
+	for _, c := range sts.Spec.Template.Spec.Containers {
+		sts.Annotations[c.Name] = c.Image
+	}
+}
```

Comment:
I think we should use one key to record the name of the container instead of many keys